### PR TITLE
Fix auto page break

### PR DESCRIPTION
**Pull request by yanyo87**
Related https://github.com/demisto/etc/issues/35023
Related https://github.com/demisto/etc/issues/30243
Related https://github.com/demisto/etc/issues/35024

1. Fix table break on table widgets & markdown 
2. show table header
3. add support for  auto page break 

old 
image

fixed 
image

### CODE REVIEW

File: src/components/Layouts/ReportLayout.js [hunk 1]
```diff
@@ -103,6 +115,22 @@ class ReportLayout extends Component {
...
+      if (item.element) {
+        clientHeight = Math.max(clientHeight, getItemHeight(item));
+      }
+    });
+
+    return clientHeight;
+  }
+
+  getItemHeight = (item) => {
+    return item.element.clientHeight || parseInt(item.element.style.height.replace('px', ''), 10);
```
Comment (Bars92) [inline src/components/Layouts/ReportLayout.js:131]: escape `item.element.style.height` if undefined

File: src/components/Layouts/ReportLayout.js [hunk 2]
```diff
@@ -68,8 +68,10 @@ class ReportLayout extends Component {
       // set dynamic height for all sections, fix top attribute.
       const itemsByRow = groupBy(Object.values(this.itemElements), item => item.gridItem.y);
       const heightMap = {};
-      Object.keys(itemsByRow).sort(compareFields).forEach((key) => {
-        const items = itemsByRow[key];
+      const rows = Object.keys(itemsByRow);
+      Object.keys(itemsByRow).sort(compareFields).forEach((key, index) => {
+        const items = itemsByRow[rows[index]];
+        const nextRowHeight = this.getMaxHeight(itemsByRow[rows[index + 1]]);
```
Comment (Bars92) [inline src/components/Layouts/ReportLayout.js:74]: shouldnt you check `rows[index + 1]` exists?

File: tests/containers/ReportContainer_spec.js [hunk 3]
```diff
@@ -451,4 +451,34 @@ describe('Report Container', () => {
...
+
+  it('Generate test template A4 layout - test auto page break', (done) => {
+    const testTemplate = TemplateProvider.getTestLayoutTemplateWithPageBreaks();
+    const dimensions = { width: 793.70078740155, height: 1122.519685039335 };
+
+    const section = prepareSections(cloneDeep(testTemplate), null, true);
+    const toRender = <ReportContainer sections={section} isLayout dimensions={dimensions} />;
+    const reportContainer = mount(toRender);
+
+    const section2 = prepareSections(cloneDeep(testTemplate), null, false);
```
Comment (Bars92) [inline tests/containers/ReportContainer_spec.js:463]:
```suggestion
    const sectionWithoutAutoPageBreak = prepareSections(cloneDeep(testTemplate), null, false);
```

File: tests/containers/ReportContainer_spec.js [hunk 4]
```diff
@@ -451,4 +451,34 @@ describe('Report Container', () => {
     expect(chartLegend.at(0).props().capitalize).to.be.true;
     expect(chartLegend.at(3).props().capitalize).to.be.false;
   });
+
+  it('Generate test template A4 layout - test auto page break', (done) => {
+    const testTemplate = TemplateProvider.getTestLayoutTemplateWithPageBreaks();
+    const dimensions = { width: 793.70078740155, height: 1122.519685039335 };
```
Comment (Bars92) [inline tests/containers/ReportContainer_spec.js:457]: perhaps move  `{ width: 210 * PIXEL_SIZE, height: 297 * PIXEL_SIZE };` to a constant

File: src/components/Layouts/ReportLayout.js (hunk 2 above)
Comment (yanyo87) [reply src/components/Layouts/ReportLayout.js:74]: no need, 
getMaxHeight handles it

**Review by Bars92 — APPROVED** [review]
👍

**Pull request merged**